test/cmdlineTests.sh

```diff
@@ -94,6 +94,56 @@ printTask "Testing unknown options..."
     fi
 )
 
+# General helper function for testing SOLC behaviour, based on file name, compile opts, exit code, stdout and stderr.
+# An failure is expected.
+test_solc_file_input_failures() {
+    local filename="${1}"
```

Comment:
Any reason to split into ``filename`` and ``solc_args``? One might just throw them together and call this function ``test_solc_output()`` instead. That function could probably be re-used throughout this whole script then.